### PR TITLE
[LANG-1615] - ArrayUtils contains && indexOf fails to handle Float.NaN

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java [hunk 1]
```diff
@@ -309,6 +309,14 @@ public void testContainsFloat() {
         assertFalse(ArrayUtils.contains(array, (float) 99));
     }
 
+    @Test
+    public void testContainsFloatNaN() {
+        float[] a = new float[] { Float.NEGATIVE_INFINITY, Float.NaN, Float.POSITIVE_INFINITY };
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java:314]: a -> array

Comment (arturobernalg) [reply src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java]: Changed

**Comment by garydgregory** [comment]
@arturobernalg please rebase on master.

**Comment by arturobernalg** [comment]
Rebase  on master, and fix checkstyle problems

**Pull request merged**